distributed/dashboard/components/nvml.py

```diff
@@ -139,13 +139,17 @@ def update(self):
                 except KeyError:
                     continue
                 metrics = ws.metrics["gpu"]
-                for j, (u, mem_used, mem_total) in enumerate(
+                for j, (u, mem_used, procs, mem_total) in enumerate(
                     zip(
                         metrics["utilization"],
                         metrics["memory-used"],
+                        metrics["procs"],
                         info["memory-total"],
                     )
                 ):
+                    # find which GPU maps to which process
+                    if ws.pid not in procs:
```

Comment:
I think I resolved the logic issues in the diagnostics display by adding in process information from each GPU.  What we had before was each worker on a node would collect all the data from the GPU(s) (if multiple were available) and pass back to this for-loop.  So we were n^2 the amount of data displayed:

Node 1-worker 1 -> GPU0->8
Node 1-worker 2 -> GPU0->7
....
Node 2-worker 1 -> GPU0->7
Node 2-worker 2 -> GPU0->7
....

This is not ideal, but by collecting process information from the GPU and passing back to the dashboard logic, we can not appropriately match up the pid of the worker with the pids reported by the GPU and only display when the pids match

Reply:
@pentschev do you think this is the best way to unique worker processes/GPU ?

Reply:
Why can't you do what I said in https://github.com/dask/distributed/pull/3810#issuecomment-701526666 ? There's only one relevant GPU per process -- the first GPU in `CUDA_VISIBLE_DEVICES`. We would need something equivalent to:

```python
def get_used_memory():
    gpu_idx = os.environ.get("CUDA_VISIBLE_DEVICES", "").split(",")[0]
    handle = pynvml.nvmlDeviceGetHandleByIndex(gpu_idx)
    return pynvml.nvmlDeviceGetMemoryInfo(handle).used

used_memory_per_gpu = client.run(get_used_memory())
```

In other words, we shouldn't be capturing data for all GPUs in `CUDA_VISIBLE_DEVICES` and reporting them all for each worker, but only the first one.

Reply:
And note that, to answer your question more directly, this may work fine until we have PID collisions, as is bound to happen in sufficiently large clusters with multiple nodes. You'd probably need to match that with a unique identifier for each node to be more resilient, not sure if we do have a way for that in Dask. But this solution is fine with me as well, the only other way it could be more reliable is to use `pynvml.nvmlDeviceGetSerial` to match by a GPUs serial number.

Reply:
What about including an IP address as well?

Reply:
sorry, forgot to push the change